File: src/ProgrammingSkills/876. Middle of the Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode midNode = head;
        int count = 1;
        int mid = count/2;
        ListNode nextNode = head;
        while(nextNode != null && nextNode.next != null){
            nextNode = nextNode.next;
            count++;
            if(mid != count/2){
                midNode = midNode.next;
                mid = count/2;
            }
        }
        return midNode;
    }
}

File: src/golang/studyPlan/DataStructure/876. Middle of the Linked List.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func middleNode(head *ListNode) *ListNode {
    var tempHead,result = head,head;
    var count = 0;
    for tempHead != nil {
        if(count % 2 == 1){
            result = result.Next;
        }
        tempHead = tempHead.Next;
        count++;

    }
    return result;
}
